src/nccl_ofi_rdma.c

```diff
@@ -3596,6 +3607,25 @@ static nccl_net_ofi_rdma_recv_comm_t *prepare_recv_comm(nccl_net_ofi_rdma_device
 	/* Add ourselves to ep's lookup array */
 	set_comm(device, r_comm->local_comm_id, &r_comm->base.base);
 
+	r_comm->control_rail.local_ep = l_comm->leader_local_ep;
```

Comment:
Sorry, just noticed this. This should be
```
r_comm->control_rail.local_ep = ep->control_rail.ofi_ep;
```

When `ENDPOINT_PER_COMM = 1`, it may be that `l_comm_ep != ep`.

Reply:
I tried the new github button to rebase the PR, turns out it does it as a merge 😞 

Good catch, will fix.

Reply:
FFR, there's a dropdown next to the button to do a rebase rather than a merge commit.